assets/js/src/newsletter_editor/blocks/products.js

```diff
@@ -107,6 +107,7 @@ Module.ProductsBlockModel = base.BlockModel.extend({
   fetchAvailableProducts: function fetchAvailableProducts() {
     var that = this;
     this.set('offset', 0);
+    this.set('contentType', 'product');
```

Comment:
Do we need it here? It should be set as default. Btw. I noticed that for other type of blocks we have also some defaults defined within `mailpoet/views/newsletter/editor.html` but not for products. So maybe adding products there could fix the issue. Not sure though.